lib/spack/spack/concretize.py

```diff
@@ -63,10 +63,11 @@ def __init__(self, abstract_spec=None):
         self._adjust_target_answer_generator = None
 
     def concretize_develop(self, spec):
+        import spack.environment as ev
```

Comment:
Why doing a local import rather than one at the top of the module?

Reply:
Because of Python 2:
```
$ SPACK_PYTHON=python2 spack --version
Traceback (most recent call last):
  File "/home/harmen/spack/bin/spack", line 96, in <module>
    import spack.main  # noqa
  File "/home/harmen/spack/lib/spack/spack/main.py", line 36, in <module>
    import spack.cmd
  File "/home/harmen/spack/lib/spack/spack/cmd/__init__.py", line 24, in <module>
    import spack.environment as ev
  File "/home/harmen/spack/lib/spack/spack/environment.py", line 20, in <module>
    import spack.concretize
  File "/home/harmen/spack/lib/spack/spack/concretize.py", line 36, in <module>
    import spack.environment as ev
AttributeError: 'module' object has no attribute 'environment'
```

There's a circularity between environment & concretize, but it's only an error on Python 2.7, not Python 3.x.

Reply:
Actually... when I drop `as ev` here, Python 2 does not complain :confused:.

Reply:
Can you open an issue as soon as we merge this, to understand what's going on?